src/System.Management.Automation/engine/interpreter/AddInstruction.cs

```diff
@@ -130,14 +130,14 @@ public static Instruction Create(Type type)
             Debug.Assert(!type.IsEnum);
             switch (type.GetTypeCode())
             {
-                case TypeCode.Int16: return s_int16 ?? (s_int16 = new AddInt16());
-                case TypeCode.Int32: return s_int32 ?? (s_int32 = new AddInt32());
-                case TypeCode.Int64: return s_int64 ?? (s_int64 = new AddInt64());
-                case TypeCode.UInt16: return s_UInt16 ?? (s_UInt16 = new AddUInt16());
-                case TypeCode.UInt32: return s_UInt32 ?? (s_UInt32 = new AddUInt32());
-                case TypeCode.UInt64: return s_UInt64 ?? (s_UInt64 = new AddUInt64());
-                case TypeCode.Single: return s_single ?? (s_single = new AddSingle());
-                case TypeCode.Double: return s_double ?? (s_double = new AddDouble());
+                case TypeCode.Int16: return s_int16 ??= new AddInt16();
+                case TypeCode.Int32: return s_int32 ??= new AddInt32();
+                case TypeCode.Int64: return s_int64 ??= new AddInt64();
+                case TypeCode.UInt16: return s_UInt16 ??= new AddUInt16();
+                case TypeCode.UInt32: return s_UInt32 ??= new AddUInt32();
+                case TypeCode.UInt64: return s_UInt64 ??= new AddUInt64();
+                case TypeCode.Single: return s_single ??= new AddSingle();
+                case TypeCode.Double: return s_double ??= new AddDouble();
```

Comment:
At some point we should go through and make these switch expressions, there's a few of these sorts of things kicking around.

Reply:
> make these switch expressions

In the code it seems it makes no sense.

Reply:
There are many switches where every code path is a `return`; these could easily be
```csharp
type.GetTypeCode() switch
{
    TypeCode.Int16 => s_int16 ??= new AddInt16(),
    //...
}
```

Etc :)

Reply:
@vexx32 Thanks, I was already considering the `IDE0066: Use switch expression` code fix for next PR.